nexus/src/external_api/http_entrypoints.rs

```diff
@@ -5048,6 +5058,201 @@ async fn updates_refresh(
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+/// View system version and update status
+#[endpoint {
+     method = GET,
+     path = "/v1/system/update/version",
+     tags = ["system"],
+}]
+async fn system_version(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+) -> Result<HttpResponseOk<SystemVersion>, HttpError> {
+    let apictx = rqctx.context();
+    let _nexus = &apictx.nexus;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        opctx.authorize(authz::Action::ListChildren, &authz::FLEET).await?;
+        Ok(HttpResponseOk(SystemVersion {
+            version_range: VersionRange {
+                low: SemverVersion::new(0, 0, 1),
+                high: SemverVersion::new(0, 0, 2),
+            },
+            status: VersionStatus::Steady {
+                reason: VersionSteadyReason::Completed,
+            },
+        }))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// View version and update status of component tree
+#[endpoint {
+     method = GET,
+     path = "/v1/system/update/components",
+     tags = ["system"],
+}]
+async fn system_component_version_list(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    query_params: Query<PaginatedById>,
+) -> Result<HttpResponseOk<ResultsPage<UpdateableComponent>>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let query = query_params.into_inner();
+    let pagparams = data_page_params_for(&rqctx, &query)?;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let components = nexus
+            .updateable_components_list_by_id(&opctx, &pagparams)
+            .await?
+            .into_iter()
+            .map(|u| u.into())
+            .collect();
+        Ok(HttpResponseOk(ScanById::results_page(
+            &query,
+            components,
+            &|_, u: &UpdateableComponent| u.identity.id,
+        )?))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// List all updates
+#[endpoint {
+     method = GET,
+     path = "/v1/system/update/updates",
+     tags = ["system"],
+}]
+async fn system_update_list(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    // TODO: pagination should probably be by most recent first, ID is nonsense
+    query_params: Query<PaginatedById>,
```

Comment:
That'll just be a sort mode we'll have to add on the ID based pagination. Pagination by ID (or name) denotes how it indicates what the next page is internally but doesn't necessarily define how it's sorted. That'll be left up to the `sort_by` (which does default to some mode). We'll just add a new sort mode to `PaginatedById` which'll cover this case.